src/main/java/core/basesyntax/MyHashMap.java

```diff
@@ -1,24 +1,100 @@
 package core.basesyntax;
 
-/**
- * <p>Реалізувати свою HashMap, а саме методи `put(K key, V value)`, `getValue()` та `getSize()`.
- * Дотриматися основних вимог щодо реалізації мапи (initial capacity, load factor, resize...)
- * За бажанням можна реалізувати інші методи інтрефейсу Map.</p>
- */
+import java.util.Objects;
+
 public class MyHashMap<K, V> implements MyMap<K, V> {
+    private static final double loadFactor = 0.75;
+    private static final int defaultCapacity = 16;
+    private Node<K, V>[] table;
+    private int size;
+    private int threshold;
+
+    public MyHashMap() {
+        this.table = new Node[defaultCapacity];
+        this.threshold = (int) (defaultCapacity * loadFactor);
+    }
 
     @Override
     public void put(K key, V value) {
-
+        Node<K, V> newNode = new Node(key, value, null);
+        int index = indexFor(key);
+        if (table[index] == null) {
+            table[index] = newNode;
+            size++;
+        } else {
+            linkNode(newNode, index);
+            if (size > threshold) {
+                resize();
+            }
+        }
     }
 
     @Override
     public V getValue(K key) {
+        int index = indexFor(key);
+        for (Node<K, V> curNode = table[index]; curNode != null; curNode = curNode.next) {
+            if (Objects.equals(key, curNode.key)) {
+                return curNode.value;
+            }
+        }
         return null;
     }
 
     @Override
     public int getSize() {
-        return 0;
+        return size;
+    }
+
+    @Override
+    public boolean isEmpty() {
+        return size == 0;
+    }
+
+    private int indexFor(K key) {
+        return key == null ? 0 : ((key.hashCode() & (table.length - 1)));
+    }
+
+    private void linkNode(Node<K, V> node, int index) {
+        Node<K, V> curNode = table[index];
+        Node<K, V> tempNode = null;
+        if (Objects.equals(node.key, curNode.key)) {
+            table[index].value = node.value;
+            return;
+        }
```

Comment:
do we need this `if` statement?

Reply:
No. I forgot to remove it from the first solution with different logic for linking.